File: qqbot-module-onebot11/src/main/java/com/mieai/qqbot/onebot11/config/OneBot11SettingsView.java
package com.mieai.qqbot.onebot11.config;

public record OneBot11SettingsView(
        String botId,
        boolean enabled,
        boolean forwardEnabled,
        String forwardBindAddress,
        Integer forwardPort,
        boolean reverseEnabled,
        String reverseUrl,
        boolean accessTokenConfigured,
        boolean heartbeatEnabled,
        int heartbeatIntervalMs,
        int reconnectIntervalMs,
        long revision) {

    static OneBot11SettingsView from(OneBot11Config value) {
        return new OneBot11SettingsView(
                value.botId().toString(),
                value.enabled(),
                value.forwardEnabled(),
                value.forwardBindAddress(),
                value.forwardPort().isPresent() ? value.forwardPort().getAsInt() : null,
                value.reverseEnabled(),
                value.reverseUrl().map(Object::toString).orElse(null),
                value.encryptedAccessToken().isPresent(),
                value.heartbeatEnabled(),
                value.heartbeatIntervalMs(),
                value.reconnectIntervalMs(),
                value.revision());
    }
}

File: qqbot-module-onebot11/src/main/kotlin/com/mieai/qqbot/onebot11/config/OneBot11SettingsView.kt
package com.mieai.qqbot.onebot11.config

data class OneBot11SettingsView(
    val botId: String,
    val enabled: Boolean,
    val forwardEnabled: Boolean,
    val forwardBindAddress: String,
    val forwardPort: Int?,
    val reverseEnabled: Boolean,
    val reverseUrl: String?,
    val accessTokenConfigured: Boolean,
    val heartbeatEnabled: Boolean,
    val heartbeatIntervalMs: Int,
    val reconnectIntervalMs: Int,
    val revision: Long,
) {
    companion object {
        fun from(value: OneBot11Config): OneBot11SettingsView = OneBot11SettingsView(
            value.botId.toString(),
            value.enabled,
            value.forwardEnabled,
            value.forwardBindAddress,
            value.forwardPort,
            value.reverseEnabled,
            value.reverseUrl?.toString(),
            value.encryptedAccessToken != null,
            value.heartbeatEnabled,
            value.heartbeatIntervalMs,
            value.reconnectIntervalMs,
            value.revision,
        )
    }
}
